src/MemoryMgmt.chpl

```diff
@@ -79,6 +89,44 @@ module MemoryMgmt {
 
         return (AutoMath.round(availableMemoryPct/100 * memAvail)*1000):uint(64);
     }
+    
+    proc getTotalMemory() : uint(64) throws {
+        var aFile = open('/proc/meminfo', ioMode.r);
+        var lines = aFile.reader().lines();
+        var line : string;
+
+        var totalMem:uint(64);
+
+        for line in lines do {
+            if line.find('MemTotal:') >= 0 {
+                var splits = line.split('MemTotal:');
+                totalMem = splits[1].strip().strip(' kB'):uint(64);
+            }
+        }
```

Comment:
should we add a break after `totalMem` is set to avoid extra iterations that are not necessary?

Reply:
@Ethan-DeBandi99 good call, added that, testing once build is complete

Reply:
@Ethan-DeBandi99 Added, please check it out.

Reply:
Looks good